### PR TITLE
QEM-14379

### PR DESCRIPTION
**Pull request by iagolivino1**
- adp text interaction test done
- add logout validation when test fails
- some improvements
- solve login_steps conflict

### CODE REVIEW

File: step_definitions/common_steps.py [hunk 1]
```diff
@@ -37,6 +39,31 @@ def get_agent_by_driver(driver_):
...
+            if outbound_camp:
+                assert tmp_agent.get('outbound_camp') == outbound_camp
+            if chat_camp:
+                assert tmp_agent.get('chat_camp') == chat_camp
+            if ready_for:
+                assert ready_for in tmp_agent.get('ready_channels')
+            agent = tmp_agent
+        except AssertionError:
+            continue
+        except AttributeError:
```
Comment (silvadf9) [inline step_definitions/common_steps.py:62]: Duplicated except clause

File: test/python/conftest.py [hunk 2]
```diff
@@ -44,6 +44,10 @@ def pytest_bdd_after_scenario(request, feature, scenario):
     print('reset variables...')
     common_steps.reset_variables()
     print('variables reset!')
+    print('checking if any agent is logged in...')
+    if request.node.__scenario_report__.current_step_report.failed:
+        login_steps.perform_logout()
```
Comment (silvadf9) [inline test/python/conftest.py:49]: We also need to add a logout mechanism for failed adapter tests

File: step_definitions/common_steps.py (hunk 1 above)
Comment (iagolivino1) [reply step_definitions/common_steps.py:62]: Actually here we are treating 2 different exceptions.
Assertion when the condition does not match
Attribute when the station_id is not set (for 'None' station option)

File: test/python/conftest.py (hunk 2 above)
Comment (iagolivino1) [reply test/python/conftest.py:49]: good point, I will work on it!

File: step_definitions/common_steps.py (hunk 1 above)
Comment (silvadf9) [reply step_definitions/common_steps.py:62]: Read it wrong

**Review by silvadf9 — APPROVED** [review]

**Pull request merged**